src/Uno.UWP/ApplicationModel/Package.Other.cs

```diff
@@ -1,106 +1,110 @@
+#nullable enable
 using System;
-using System.Collections.Generic;
 using System.Reflection;
-using System.Security.Cryptography;
 using System.Xml;
-using Uno.Extensions;
 using Uno.Foundation.Logging;
-using Uno.UI;
-using Windows.ApplicationModel.Core;
-using Windows.ApplicationModel.Email.DataProvider;
-using Windows.Storage;
 
-namespace Windows.ApplicationModel
+namespace Windows.ApplicationModel;
+
+public partial class Package
 {
-	public partial class Package
-	{
-		private const string PackageManifestName = "Package.appxmanifest";
-		private static Assembly? _entryAssembly;
-		private string _displayName = "";
-		private string _logo = "ms-appx://logo";
-		private bool _manifestParsed;
+	private const string PackageManifestName = "Package.appxmanifest";
 
-		private bool GetInnerIsDevelopmentMode() => false;
+	private static Assembly? _entryAssembly;
 
-		private DateTimeOffset GetInstallDate() => DateTimeOffset.Now;
+	partial void InitializePlatform()
+	{
+	}
 
-		private string GetInstalledPath()
-		{
-			if (_entryAssembly?.Location is { Length: > 0 } location)
-			{
-				return global::System.IO.Path.GetDirectoryName(location) ?? "";
-			}
-			else if (AppContext.BaseDirectory is { Length: > 0 } baseDirectory)
-			{
-				return global::System.IO.Path.GetDirectoryName(baseDirectory) ?? "";
-			}
+	internal static bool IsManifestInitialized { get; private set; }
 
-			return Environment.CurrentDirectory;
-		}
+	private bool GetInnerIsDevelopmentMode() => false;
+
+	private DateTimeOffset GetInstallDate() => DateTimeOffset.Now;
 
-		public string DisplayName
+	private string GetInstalledPath()
+	{
+		if (_entryAssembly?.Location is { Length: > 0 } location)
 		{
-			get
-			{
-				TryParsePackageManifest();
-				return _displayName;
-			}
+			return global::System.IO.Path.GetDirectoryName(location) ?? "";
 		}
-
-		public Uri Logo
+		else if (AppContext.BaseDirectory is { Length: > 0 } baseDirectory)
 		{
-			get
-			{
-				TryParsePackageManifest();
-				return new Uri(_logo, UriKind.RelativeOrAbsolute);
-			}
+			return global::System.IO.Path.GetDirectoryName(baseDirectory) ?? "";
 		}
 
-		internal static void SetEntryAssembly(Assembly entryAssembly)
+		return Environment.CurrentDirectory;
+	}
+
+	public string DisplayName { get; private set; } = "";
+
+	public Uri Logo { get; private set; } = new Uri("ms-appx://logo", UriKind.RelativeOrAbsolute);
+
+	internal static void SetEntryAssembly(Assembly entryAssembly)
+	{
+		_entryAssembly = entryAssembly;
+		Current.Id.Name = entryAssembly.GetName().Name; // Set the package name to the entry assembly name by default.
+		Current.ParsePackageManifest();
+		IsManifestInitialized = true;
+	}
+
+	private void ParsePackageManifest()
+	{
+		if (_entryAssembly is null)
 		{
-			_entryAssembly = entryAssembly;
+			return;
 		}
 
-		private void TryParsePackageManifest()
+		var manifest = _entryAssembly.GetManifestResourceStream(PackageManifestName);
+
+		if (manifest is not null)
```

Comment:
nit nit: saving one allocation
```suggestion
		if (_entryAssembly.GetManifestResourceStream(PackageManifestName) is {} manifest)
```